pkg/util/timeutil/ptp/ptp_clock_linux.go

```diff
@@ -62,19 +77,29 @@ func MakeClock(ctx context.Context, clockDevicePath string) (Clock, error) {
 		clockID,
 	)
 	var ts C.struct_timespec
-	_, err = C.clock_gettime(C.clockid_t(clockID), &ts)
+	if p.cGetTime == nil {
+		_, err = C.clock_gettime(C.clockid_t(clockID), &ts)
+	} else {
+		err = p.cGetTime(C.clockid_t(clockID), &ts)
+	}
+
 	if err != nil {
-		return result, errors.Wrap(err, "UseClockDevice: error calling clock_gettime")
+		return p, errors.Wrap(err, "UseClockDevice: error calling clock_gettime")
 	}
-	result.clockDeviceID = clockID
+	p.clockDeviceID = clockID
 
-	return result, nil
+	return p, nil
 }
 
 // Now implements the hlc.WallClock interface.
-func (p Clock) Now() time.Time {
+func (p *Clock) Now() time.Time {
 	var ts C.struct_timespec
-	_, err := C.clock_gettime(C.clockid_t(p.clockDeviceID), &ts)
+	var err error
+	if p.cGetTime == nil {
+		_, err = C.clock_gettime(C.clockid_t(p.clockDeviceID), &ts)
+	} else {
+		err = p.cGetTime(C.clockid_t(p.clockDeviceID), &ts)
+	}
```

Comment:
Since `Now()` is on the critical path to many things, it would be nice to avoid branching here. A couple of options:

1. Never have `p.cGetTime == nil`, and call it unconditionally. It would need to be initialized in `MakeClock()`.
2. Hide the test-only branch behind `if buildutil.CrdbTestBuild` which is more likely to be optimized out in non-test builds.

(2) is preferred.